src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.AspNetCore.Mvc;
+using Esfa.Recruit.Qa.Web.Configuration.Routing;
+using System.Threading.Tasks;
+using Esfa.Recruit.QA.Web.Orchestrators;
+using Esfa.Recruit.Qa.Web.Security;
+using Microsoft.AspNetCore.Authorization;
+using Esfa.Recruit.QA.Web.ViewModels.ManageProvider;
+using Esfa.Recruit.Qa.Web.Configuration;
+using Esfa.Recruit.Qa.Web.Extensions;
+
+namespace Esfa.Recruit.QA.Web.Controllers
+{
+    [Authorize(Policy = AuthorizationPolicyNames.TeamLeadUserPolicyName)]
+    [Route(RoutePaths.UnBlockOrganisationRoutePath)]
+    public class UnBlockOrganisationController : Controller
+    {
+        private readonly UnBlockOrganisationOrchestrator _orchestrator;
+
+        public UnBlockOrganisationController(UnBlockOrganisationOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+        }
+
+        [HttpGet("", Name = RouteNames.UnBlockProvider_Confirm_Get)]
+        public async Task<IActionResult> RestoreAccess(string organisationId)
+        {
+            if (long.TryParse(organisationId, out var ukprn) == false) return BadRequest();
+            var vm = await _orchestrator.GetBlockedOrganisationViewModel(ukprn);
+            return View(vm);
+        }
+
+        [HttpPost("", Name = RouteNames.UnBlockProvider_Confirm_Post)]
+        public async Task<IActionResult> RestoreAccess(UnBlockTrainingProviderEditModel model)
+        {
+            if (ModelState.IsValid == false)
+            {
+                return View(new UnBlockTrainingProviderEditModel() { Ukprn = model.Ukprn, ProviderName = model.ProviderName });
```

Comment:
Naming is a bit wonky between action names, view models and views. Try and keep them aligned if possible.

Can we change usages of the word `RestoreAccess` to `Unblock`.  The front-end can say 'Restore Access' but I think in the code keeping to block/unblock is easier to understand. 

Change the ViewModel from `UnBlockTrainingProviderEditModel` to `ConfirmTrainingProviderUnblockingViewModel` so it aligns with the existing `ConfirmTrainingProviderBlockingViewModel`.

Change this Function name to `ConfirmTrainingProviderBlocking`

Change the view from `RestoreAccess.cshtml` to `ConfirmTrainingProviderUnblocking.cshtml`.

...etc

Reply:
👍